src/oncify.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Returns a function that runs only once
+ *
+ * @param {Function}  fn    the function to be made to run only once
+ *
+ * @returns {Function}      the function that runs only once
+ */
+const oncify = function (fn: Function): Function {
+  if (typeof fn !== "function") {
+    throw TypeError(
+      `Argument of type ${typeof fn} is not assignable to parameter of type 'Function'`
+    );
+  }
+  let hasRun = false;
+  return (...args: any) => {
+    if (hasRun === false) {
+      hasRun = true;
+      return fn(...args);
+    } else {
+      return undefined;
```

Comment:
If you remove the `else` block, `undefined` will be returned implicitly.

Reply:
Yup! I was trying to be explicit for the reader of the code. That's three changes to make